attestation-evidence-broker/docs/design/design.md

```diff
@@ -0,0 +1,122 @@
+
+SEV(-ES) runtime attestation includes two parties: attester and verifier.
+
+- The attester running in a SEV(-ES) guest is responsible for collecting attestation evidence.
+- The verifier running in a trusted environment is responsible for verifying the attestation evidence provided by the attester. Note that **the verifier can run on any platform even non-TEE**.
+
+# SEV(-ES) Attestation Evidence
+
+Verifier uses the following SEV(-ES) certificate chains to verify the signature of the attestation report generated by the `ATTESTATION` command [2]:
+
+```bash
+certificate chain for device identity: ARK -> ASK -> CEK -> PEK -> report
+certificate chain for platform owner identity: OCA -> PEK -> report
+```
+
+- `foo -> bar` indicates using the public key of `foo` to verify the signature of `bar`.
+- ARK is the root of trust of AMD and OCA is the root of trust for platform owners. OCA has two ways:
+    1. Self-owned: The OCA Key Pair and self-signed OCA certificate are automatically generated by the SEV(-ES) firmware.
+    2. Externally-owned: External users use OCA Key Pair to generate self-signed OCA certificates in a trusted environment.
+
+Verifier needs to verify the attestation report with the certificate chain. ARK and ASK can be obtained directly from the AMD KDS server. CEK, PEK, OCA, and attestation report are related to the specific SEV(-ES) platform, therefore **SEV(-ES) Attestation Evidence collected by attester should include attestation report, PEK, CEK, and OCA certificate. **
+
+| Contents of SEV(-ES) Attestation Evidence | SEV(-ES) firmware command |
+| :-: | :-: |
+| attestation report                        | ATTESTATION               |
+| CEK                                       | GET_ID                    |
+| OCA,PEK                                   | PDH_CERT_EXPORT           |
+
+## The methods of query SEV(-ES) Attestation Evidence
+
+According to the [officially feedback](https://github.com/AMDESE/AMDSEV/issues/71#issuecomment-926118314), **SEV(-ES) firmware APIs don't support query attestation report in SEV(-ES) guest and there is no plan to support it in the future. Instead, this capability will be available in SEV-SNP.**
+
+In some scenarios, the guest application needs to query the attestation report to establish a trusted channel with the remote peer. There are two approaches for a guest application to query an attestation evidence:
+
+- Hypercall approach
+- Vsock approach
+
+Considering time and cost, we only need to implement one of them.
+
+### Hypercall approach
+
+SEV(-ES) guest exits to VMM using `hypercall` and then interacts with PSP firmware to query attestation evidence, including attestation report, PEK, CEK, OCA certificate. In general, it includes the following steps:
```

Comment:
s/to query attestation evidence/to query the components composing an attestation evidence/

Reply:
s/In general, it includes the following steps:/To build an attestation evidence, the steps include:/